extension/sqlsmith/fuzzyduck.cpp

```diff
@@ -62,18 +65,29 @@ void FuzzyDuck::FuzzAllFunctions() {
 	EndFuzzing();
 }
 
-string FuzzyDuck::GenerateQuery() {
-	LogTask("Generating query with seed " + to_string(seed));
+idx_t FuzzyDuck::GenerateRandomNumber() {
 	auto &engine = RandomEngine::Get(context);
-	// set the seed
-	engine.SetSeed(seed);
-	// get the next seed
-	seed = engine.NextRandomInteger();
+	return engine.NextRandomInteger() % 1000;
+}
 
+string FuzzyDuck::GenerateQuery() {
 	// generate the statement
 	StatementGenerator generator(context);
-	auto statement = generator.GenerateStatement();
-	return statement->ToString();
+	// accumulate statement(s)
+	auto statement = string("");
+	if (generator.RandomPercentage(10)) {
+		// multi statement
+		idx_t number_of_statements = GenerateRandomNumber();
```

Comment:
You could also make `idx_t RandomValue(idx_t max);` in StatementGenerator a public function. Then you don't have to create a random number generator here.